file.go

```diff
@@ -178,3 +181,57 @@ func (k *fileKeyring) Keys() ([]string, error) {
 
 	return keys, nil
 }
+
+func (k *fileKeyring) getNewPassword() error {
+	dir, err := k.resolveDir()
+	if err != nil {
+		return err
+	}
+
+	if k.newPassword == "" {
+		pwd, err := k.newPasswordFunc(fmt.Sprintf("Enter new passphrase for %q", dir))
+		if err != nil {
+			return err
+		}
+		k.newPassword = pwd
+	}
+	return nil
+}
+
+func (k *fileKeyring) Changepw(key string) error {
```

Comment:
Also here, to avoid code duplication, you could do something like:
```
// Get item with current pswd
item := k.GetItem()
// Ask for new pswd and update k.password
...
// Store item
k.SetITem(item)
```

Reply:
The problem with this approach is the prompts for the old and new password look the same.

The other way is for static prompts to be placed in this keyring package like "Enter new password", but then it could not be customized when called from cosmos-sdk.

Is it better to hard code old and new password prompts in cosmos/keyring? Or have a single prompt in cosmos-sdk and additional prompts for old and new password in other places within cosmos-sdk or cosmos/keyring?

Reply:
check that `TerminalPrompt` receives as argument the prompt you desire, so you can re-use the current instance of `FilePasswordFunc`

Reply:
just a nit, consider using `GetItem() `and `SetItem()` to avoid code duplication

Reply:
Using SetItem in this case makes it harder to understand when testing.
The keyring config no longer has newPassword (removed to respond to code comment), and password needs to be overwritten by newPassword to SetItem.
Then the keyring needs to have newPassword overwritten again by the oldPassword for each key in the key ring to unlock the next key for GetItem.

Having both the old and new password is helpful when changing the password for many keys in a shared-password cosmos-sdk keyring.
It eliminatees re-prompting for the old password and the new password for each key 
We can make cosmos-sdk responsible for fixing the overwritten password after each key gets a new password

Should we bring back the fields we just removed?